Applications/ApplicationsLib/UncoupledProcessesTimeLoop.h

```diff
@@ -225,8 +224,7 @@ class UncoupledProcessesTimeLoop
 };
 
 //! Builds an UncoupledProcessesTimeLoop from the given configuration.
-template<typename Matrix, typename Vector>
-std::unique_ptr<UncoupledProcessesTimeLoop<Matrix, Vector> >
+inline std::unique_ptr<UncoupledProcessesTimeLoop>
```

Comment:
Could you please move all the now "inlined" methods to a cpp file rather than using `inline`? In this particular case there won't be much advantage from that, apart from the separation of declaration/implementation. But in general this separation can lead to shorter compile times.